resources/metadata/golangBuild.yaml

```diff
@@ -22,6 +22,16 @@ spec:
           - PARAMETERS
           - STAGES
           - STEPS
+      - name: buildSettingsInfo
+        type: string
+        description: build settings info is typically filled by the step automatically to create information about the build settings that were used during the maven build . This information is typically used for compliance related processes.
+        scope:
+          - STEPS
+          - STAGES
+          - PARAMETERS
+        resourceRef:
+          - name: commonPipelineEnvironment
```

Comment:
for multiple build we should enhance the existing one, shouldn't we?
I essentially copied this from `mavenBuild`

Do I overlook s.th.?

Reply:
you are right , makes sense for a multiple build scenario , if we call the step again (2/more go builds in the same run) , we also then need to enhance (thinking out loud if we should handle it like below then?)

```
buildSettingsInfo, err := buildsettings.CreateBuildSettingsInfo(&buildConfig, stepName)`
if len(config.buildSettingsInfo) > 0
 config.buildSettingsInfo.append(config.buildSettingsInfo.append, buildSettingsInfo)
 commonPipelineEnvironment.custom.buildSettingsInfo = config.buildSettingsInfo
else
 commonPipelineEnvironment.custom.buildSettingsInfo = buildSettingsInfo
```

so that we dont loos the previous settings ? 

you are correct, would need to do this for all build tools

Reply:
yup makes sense, may need to adapt , how do we adapt if there is already a buildSettingsInfo from a previous build step , but for time being lets keep it